common/executor.js

```diff
@@ -2,22 +2,23 @@
 
 var _ = require('underscore');
 var async = require('async');
-var exec = require('child_process').exec;
 var fs = require('fs');
 var path = require('path');
 var poller = require('./poller.js');
+var spawn = require('child_process').spawn;
```

Comment:
This allows us to listen to events instead of callbacks which is needed for this approach.